homeassistant/components/srp_energy/config_flow.py

```diff
@@ -1,66 +1,106 @@
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    _LOGGER.debug("Validate User Input with client")
+    srp_client = SrpEnergyClient(
+        data[CONF_ID],
+        data[CONF_USERNAME],
+        data[CONF_PASSWORD],
+    )
 
-                srp_client = SrpEnergyClient(
-                    user_input[CONF_ID],
-                    user_input[CONF_USERNAME],
-                    user_input[CONF_PASSWORD],
-                )
+    is_valid = await hass.async_add_executor_job(srp_client.validate)
 
-                is_valid = await self.hass.async_add_executor_job(srp_client.validate)
+    _LOGGER.debug("Is valid User Input with client: %s", is_valid)
+    if not is_valid:
+        raise InvalidAuth
 
-                if is_valid:
-                    return self.async_create_entry(
-                        title=user_input[CONF_NAME], data=user_input
-                    )
+    return is_valid
 
-                errors["base"] = "invalid_auth"
 
-            except ValueError:
-                errors["base"] = "invalid_account"
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle an SRP Energy config flow."""
+
+    VERSION = 1
 
+    @callback
+    def _show_form(self, errors: dict[str, Any] | None = None) -> FlowResult:
+        """Show the form to the user."""
+        _LOGGER.debug("Show Form")
         return self.async_show_form(
-            step_id="user", data_schema=vol.Schema(self.config), errors=errors
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_ID): str,
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Optional(
+                        CONF_NAME, default=self.hass.config.location_name
+                    ): str,
+                    vol.Optional(CONF_IS_TOU, default=False): bool,
+                }
+            ),
+            errors=errors or {},
         )
 
-    async def async_step_import(self, import_config):
-        """Import from config."""
-        # Validate config values
-        return await self.async_step_user(user_input=import_config)
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        _LOGGER.debug("Config entry")
+        if not user_input:
+            return self._show_form()
+
+        errors = {}
+
+        try:
+            await validate_input(self.hass, user_input)
+        except ValueError:
+            errors["base"] = "invalid_account"
+            return self._show_form(errors)
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+            return self._show_form(errors)
```

Comment:
Instead of two exact lines, you could collect errors in the `errors` dict, then show the form _after_ the try/except block (if `errors` is not empty).